### PR TITLE
add reset-host command

### PR DESCRIPTION
**Pull request by dhellmann**
Add a command to reset the host status for dev testing

### CODE REVIEW

**Comment by dhellmann** [comment]
The  latest draft of this PR also sets up the client code generator and adds client code to the repo.

/cc @maelk @russellb @honza @kirankt

**Comment by dhellmann** [comment]
I have pushed the image referenced in `hack/client-generate.sh` built from `hack/Dockerfile.code-generator` to https://quay.io/repository/metal3-io/client-code-generator and will set up the build trigger and the CI job that uses the image when all of this merges.

**Comment by kirankt** [comment]
@dhellmann Can you rebase? Looks like there is a conflict in Makefile.

File: cmd/reset-host/main.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		}
+
+		if *wipeAnnotation {
+			annotations := host.GetAnnotations()
+			if annotations != nil {
+				if _, ok := annotations[metal3v1alpha1.StatusAnnotation]; ok {
+					delete(annotations, metal3v1alpha1.StatusAnnotation)
+				}
+
+				fmt.Printf("annotation... ")
```
Comment (kirankt) [inline cmd/reset-host/main.go:65]: Can this print be more descriptive? Perhaps "deleting annotation"?

Comment (dhellmann) [reply cmd/reset-host/main.go:65]: The output for each host ends up all on one line, like:

```
starting "hostname" annotation... status... done
```

when the option to clear the annotation isn't set, it is

```
starting "hostname" status... done
```

**Comment by dhellmann** [comment]
rebased

**Pull request closed without merging**